packages/fxa-settings/src/lib/metrics.ts

```diff
@@ -0,0 +1,331 @@
+
+import sentryMetrics from 'fxa-shared/lib/sentry';
+
+const NOT_REPORTED_VALUE = 'none';
+const UNKNOWN_VALUE = 'unknown';
+
+type Optional<T> = T | typeof NOT_REPORTED_VALUE;
+
+type ScreenInfo = {
+  clientHeight: Optional<number>;
+  clientWidth: Optional<number>;
+  devicePixelRatio: Optional<number>;
+  height: Optional<number>;
+  width: Optional<number>;
+};
+
+type EventSet = {
+  offset: number;
+  type: string;
+};
+
+type ExperimentGroup = {
+  choice: string;
+  group: string;
+};
+
+type MarketingCampaign = {
+  campaignId: string;
+  clicked: boolean;
+  url: string;
+};
+
+type UserPreferences = {
+  [userPref: string]: boolean;
+};
+
+type ConfigurableProperties = {
+  experiments: ExperimentGroup[];
+  lang: string | typeof UNKNOWN_VALUE;
+  marketing: MarketingCampaign[];
+  newsletters: string[];
+  startTime: number;
+  uid: Optional<string>;
+  userPreferences: UserPreferences;
+  utm_campaign: Optional<string>;
+  utm_content: Optional<string>;
+  utm_medium: Optional<string>;
+  utm_source: Optional<string>;
+  utm_term: Optional<string>;
+};
+
+type EventData = FlowQueryParams &
+  ConfigurableProperties & {
+    duration: number;
+    events: EventSet[];
+    flushTime: number;
+    referrer: string;
+    screen: ScreenInfo;
+    [additionalProperty: string]: any;
+  };
+
+let initialized = false;
+let viewNamePrefix: string | null;
+let flowEventData: FlowQueryParams;
+let configurableProperties: ConfigurableProperties = defaultConfigProps();
+
+function defaultConfigProps(): ConfigurableProperties {
+  return {
+    experiments: [],
+    lang: UNKNOWN_VALUE,
+    marketing: [],
+    newsletters: [],
+    // TODO: performance.timing.navigationStart should work, but doesn't
+    startTime: 123,
+    uid: NOT_REPORTED_VALUE,
+    userPreferences: {},
+    utm_campaign: NOT_REPORTED_VALUE,
+    utm_content: NOT_REPORTED_VALUE,
+    utm_medium: NOT_REPORTED_VALUE,
+    utm_source: NOT_REPORTED_VALUE,
+    utm_term: NOT_REPORTED_VALUE,
+  };
+}
+
+/**
+ * Get user's window/screen info
+ */
+function getScreenInfo(): ScreenInfo {
+  const documentElement = window.document.documentElement || {};
+  const screen = window.screen || {};
+
+  return {
+    clientHeight: documentElement.clientHeight || NOT_REPORTED_VALUE,
+    clientWidth: documentElement.clientWidth || NOT_REPORTED_VALUE,
+    devicePixelRatio: window.devicePixelRatio || NOT_REPORTED_VALUE,
+    height: screen.height || NOT_REPORTED_VALUE,
+    width: screen.width || NOT_REPORTED_VALUE,
+  };
+}
+
+/**
+ * Send metrics data to the content-server metrics
+ * endpoint via navigator.sendBeacon
+ *
+ * @param eventData
+ */
+function postMetrics(eventData: EventData) {
+  if (!initialized || !window.navigator.sendBeacon) {
+    return;
+  }
+
+  window.navigator.sendBeacon('/metrics', JSON.stringify(eventData));
+}
+
+/**
+ * Reset Metrics setup; used for testing
+ */
+export function reset() {
+  initialized = false;
+  viewNamePrefix = null;
```

Comment:
There is a ton of hidden complexity in the `viewNamePrefix`, and it was mostly used for flows involving signin/signup. I think its ok to leave leave this since it would be nice to refactor away from this.

Reply:
Yeah, there's definitely a lot going on in that existing metrics lib 😄 Just to confirm, you think we can leave it like this? Or maybe I should remove it since we don't deal with signup/in flows in Settings?